pimd/pim_sock.c

```diff
@@ -179,6 +179,49 @@ static inline int pim_setsockopt(int protocol, int fd, struct interface *ifp)
 }
 #endif
 
+int pim_reg_sock(void)
+{
+	int fd;
+	long flags;
+
+	frr_with_privs (&pimd_privs) {
+		fd = socket(AF_INET, SOCK_RAW, IPPROTO_RAW);
+	}
+
+	if (fd < 0) {
+		zlog_warn("Could not create raw socket: errno=%d: %s", errno,
+			  safe_strerror(errno));
+		return PIM_SOCK_ERR_SOCKET;
+	}
+
+	if (sockopt_reuseaddr(fd)) {
+		zlog_warn(
```

Comment:
This is already printed as flog_err, do we need to print it as a warning as well?

Reply:
The lib does not print the errno details, therefore I kept it here.

Reply:
I will move the errno details to the lib and remove it from here.